plugins/serverid/plugin.go

```diff
@@ -92,8 +92,8 @@ func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
 		log.Infof("requested server ID does not match this server's ID. Got %v, want %v", req.ServerIPAddr, v4ServerID)
 		return nil, true
 	}
-	resp.ServerIPAddr = make(net.IP, net.IPv4len)
-	copy(resp.ServerIPAddr[:], v4ServerID)
+	resp.ServerIPAddr = make(net.IP, len(v4ServerID))
```

Comment:
Is this change needed after the change in line 113?

Reply:
probably not, but it looks more correct anyway

Reply:
probably not, but it looks more correct anyway

Reply:
It doesn't look more correct to me: you have to know that somewhere else in the code, something ensures the address is 4 bytes.
If you also call To4() here in addition to the other place, it's more obvious that we're not putting an ipv6 address in an ipv4 variable

(This would be less of an issue if go had separate types for v4-only addresses, but it doesn't)

Reply:
it's more correct because that change actually fixes the bug too.
 I am not very familiar with go, but I don't get the [:] either.

Reply:
I would stick to having `net.IPv4len` here and only change line 113.

Reply:
oh well, who cares